src/i18n/I18nService.ts

```diff
@@ -4,35 +4,70 @@ import LanguageDetector from 'i18next-browser-languagedetector';
 import * as H from 'history';
 
 import * as PathUtils from '../common/reactRouterWithLanguageSupport/pathUtils';
+import Config from '../config';
+import Logger from '../logger';
 import en from './en.json';
 import fi from './fi.json';
 import sv from './sv.json';
 import setYupLocale from './setYupLocale';
 
-const supportedLanguages = ['fi', 'sv', 'en'];
-export type Language = typeof supportedLanguages[number];
+const defaultLanguages = ['fi', 'sv', 'en'];
+export type Language = string;
```

Comment:
I had to use a more general type because we don't have knowledge about which languages may be configured.